src/status_im/subs/wallet/wallet.cljs

```diff
@@ -179,21 +179,46 @@
                                   :type                :default
                                   :name                name
                                   :address             address}
+                  :blur?         true
                   :networks      networks
                   :state         :default
                   :action        :none})))))
 
+(defn- format-settings-missing-keypair-accounts
+  [accounts]
+  (->> accounts
+       (map (fn [{:keys [customization-color emoji]}]
+              {:customization-color customization-color
+               :emoji               emoji
+               :type                :default}))))
+
 (rf/reg-sub
  :wallet/settings-keypairs-accounts
  :<- [:wallet/keypairs]
- (fn [keypairs [_ format-options]]
-   (->> keypairs
-        (map (fn [{:keys [accounts name type key-uid]}]
-               {:type     (keyword type)
-                :name     name
-                :key-uid  key-uid
-                :accounts (format-settings-keypair-accounts accounts format-options)})))))
-
+ :<- [:wallet/accounts]
+ (fn [[keypairs accounts] [_ format-options]]
+   (let [grouped-accounts      (->> accounts
```

Comment:
Initially I hoped to just iterate over all the key-pairs and accounts, but it seemed that key-pairs were not immediately synced with the accounts after creating a new account and key-pair, but accounts seemed to be updated immediately. However, that may have been me discovering a bug (?), because it seems I need to sometimes re-login to re-query the existing key-pairs and accounts.

Maybe I can try to fix that here, or I can try to create a bug issue to investigate. Thoughts?

Reply:
I guess I know the problem. From the video in the PR description. The user is not logged in (in Device A - mobile or desktop) and creates a new key pair in other device (Device B - mobile or desktop).

Usually, the sync message (in our case, new key pair details) is received after the user logs in. But, we request the accounts and key pair data as soon as the user logs in and whatever data is available in the status-go is sent back to the client. When the sync message is received (in Device A), it will add those details in the db (in status-go) and notify the client via signal. We need to handle that part.

I will add that support as I'm working on something related to that.

In short, it's not a bug. It's feature needs to be built 😄

Reply:
The existing code/PR is good to go. We can clean up and refactor later once we handle all those cases (regarding key pairs and accounts). 👍